applications/FluidDynamicsApplication/custom_utilities/fluid_element_data.cpp

```diff
@@ -111,6 +113,32 @@ void FluidElementData<TDim, TNumNodes, TElementIntegratesInTime>::FillFromHistor
     }
 }
 
+template <size_t TDim, size_t TNumNodes, bool TElementIntegratesInTime>
+void FluidElementData<TDim, TNumNodes, TElementIntegratesInTime>::FillFromNonHistoricalNodalData(
+    NodalScalarData& rData,
+    const Variable<double>& rVariable,
+    const Geometry<Node<3>>& rGeometry)
+{
+    noalias(rData) = ZeroVector(TNumNodes);
+    for (size_t i = 0; i < TNumNodes; i++) {
+        rData[i] = rGeometry[i].GetValue(rVariable);
+    }
+}
+
+template <size_t TDim, size_t TNumNodes, bool TElementIntegratesInTime>
+void FluidElementData<TDim, TNumNodes, TElementIntegratesInTime>::FillFromNonHistoricalNodalData(
+    NodalVectorData& rData,
+    const Variable<array_1d<double, 3>>& rVariable,
+    const Geometry<Node<3>>& rGeometry)
+{
+    for (size_t i = 0; i < TNumNodes; i++) {
+        const array_1d<double, 3>& r_nodal_values = rGeometry[i].GetValue(rVariable);
+        for (size_t j = 0; j < rData.size2(); j++) {
```

Comment:
Hi, Wouldn't this be easier to the compiler to unroll if `TDim` is used instead of `rData.size2()` because `rData.size2()` is always `TDim` ryt?

Reply:
Agree, I also think it would eventually help the optimizer. However, I'm not sure at all if `rData.size2()` could be 3 even in 2D cases (plenty of zeroes). Probably we did it in this way because of this but we need to revisit it before changing it.